lib/iex/lib/iex/evaluator.ex

```diff
@@ -253,7 +253,7 @@ defmodule IEx.Evaluator do
       if path do
         [path]
       else
-        Enum.map([".iex.exs", "~/.iex.exs"], &Path.expand/1)
+        Enum.map([".iex.exs", System.get_env("IEX_HOME", "~/.iex.exs")], &Path.expand/1)
```

Comment:
After reading the docs, I just realized this is a bit different from MIX_HOME. MIX_HOME points to a directory where we would find files, this is pointing to a file. I think you might want this:

```suggestion
        Enum.map([".", System.get_env("IEX_HOME", "~")], fn dir ->
          dir |> Path.join(".iex.exs") |> Path.expand()
        end)
```